### DISCUSSION TITLE
Object in Phalcon\Paginator\Adapter\Model Pagination

### DISCUSSION THREAD

**Discussion by leonardobravi** [phalcon/cphalcon#15387] [Q&A] [answered]
I want to use `Phalcon\Paginator\Adapter\Model` for models pagination
but I need to have the real object in items not the `->toArray()` representation.
The first question is why `Phalcon\Paginator\Adapter\Model` use array instead of Models?
Furthermore there is a way to obtain a real resultset?

Tnx

**Comment by Jeckerson** [comment, marked as answer]
Hello

Related - https://github.com/phalcon/cphalcon/issues/15074

Will be fixed in `v5.0`. For version `v4.0` you will need to extend `Model` class and rewrite. Here is ready to use example:
https://gist.github.com/Jeckerson/08be8255c541fca381d9c32713916f1c

**Reply by leonardobravi** [reply]
Thanks I'll try it now
